src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -3340,17 +3340,18 @@ private DnsNameRepresentation GetDnsNameRepresentation(string dnsName)
         /// </summary>
         public DnsNameProperty(X509Certificate2 cert)
         {
-            _dnsList = new List<DnsNameRepresentation>();
-
-            // extract DNS name from subject distinguish name
-            // if it exists and does not contain a comma
-            // a comma, indicates it is not a DNS name
-            if (cert.Subject.StartsWith(distinguishedNamePrefix, StringComparison.OrdinalIgnoreCase) &&
-                !cert.Subject.Contains(','))
+            // Extract DNS name from Subject distinguished name
+            foreach (X500RelativeDistinguishedName distinguishedName in cert.SubjectName.EnumerateRelativeDistinguishedNames())
             {
-                string parsedSubjectDistinguishedDnsName = cert.Subject.Substring(distinguishedNamePrefix.Length);
-                DnsNameRepresentation dnsName = GetDnsNameRepresentation(parsedSubjectDistinguishedDnsName);
-                _dnsList.Add(dnsName);
+                // Check distinguished name has single element and has Common Name (CN) OID
+                if (!distinguishedName.HasMultipleElements &&
+                     distinguishedName
+                        .GetSingleElementType().Value
+                        .Equals(CommonNameOid, StringComparison.OrdinalIgnoreCase))
+                {
+                    DnsNameRepresentation dnsName = GetDnsNameRepresentation(distinguishedName.GetSingleElementValue());
```

Comment:
GetSingleElementValue() could return `null`. This would mean a malformed certificate common name, nevertheless it would be safe to skip `null`.

Reply:
@iSazonov Good point, fixed in 6dec6f4fb4a57a7e09176732c0942221b88f562c. I added checks for both OID and element value since they are both `string?` nullable types.